pkg/agent/flowexporter/connections/connections.go

```diff
@@ -1,157 +1,121 @@
 func NewConnectionStore(
-	connTrackDumper ConnTrackDumper,
-	flowRecords *flowrecords.FlowRecords,
 	ifaceStore interfacestore.InterfaceStore,
-	v4Enabled bool,
-	v6Enabled bool,
 	proxier proxy.Proxier,
-	npQuerier querier.AgentNetworkPolicyInfoQuerier,
-	pollInterval time.Duration,
+	isDenyConn bool,
 ) *connectionStore {
 	return &connectionStore{
-		connections:          make(map[flowexporter.ConnectionKey]*flowexporter.Connection),
-		flowRecords:          flowRecords,
-		connDumper:           connTrackDumper,
-		ifaceStore:           ifaceStore,
-		v4Enabled:            v4Enabled,
-		v6Enabled:            v6Enabled,
-		antreaProxier:        proxier,
-		networkPolicyQuerier: npQuerier,
-		pollInterval:         pollInterval,
+		connections:   make(map[flowexporter.ConnectionKey]*flowexporter.Connection),
+		ifaceStore:    ifaceStore,
+		antreaProxier: proxier,
+		isDenyConn:    isDenyConn,
 	}
 }
 
-// Run enables the periodical polling of conntrack connections at a given flowPollInterval.
-func (cs *connectionStore) Run(stopCh <-chan struct{}) {
-	klog.Infof("Starting conntrack polling")
-
-	pollTicker := time.NewTicker(cs.pollInterval)
-	defer pollTicker.Stop()
+// GetConnByKey gets the connection in connection map given the connection key.
+func (cs *connectionStore) GetConnByKey(connKey flowexporter.ConnectionKey) (*flowexporter.Connection, bool) {
+	cs.mutex.Lock()
+	defer cs.mutex.Unlock()
+	conn, found := cs.connections[connKey]
+	return conn, found
+}
 
-	for {
-		select {
-		case <-stopCh:
-			break
-		case <-pollTicker.C:
-			_, err := cs.Poll()
-			if err != nil {
-				// Not failing here as errors can be transient and could be resolved in future poll cycles.
-				// TODO: Come up with a backoff/retry mechanism by increasing poll interval and adding retry timeout
-				klog.Errorf("Error during conntrack poll cycle: %v", err)
-			}
-			// AddOrUpdateFlowRecord method does not return any error, hence no error handling required.
-			cs.ForAllConnectionsDo(cs.flowRecords.AddOrUpdateFlowRecord)
-			klog.V(2).Infof("Flow records are successfully updated")
+// ForAllConnectionsDo execute the callback for each connection in connection map.
+func (cs *connectionStore) ForAllConnectionsDo(callback flowexporter.ConnectionMapCallBack) error {
+	cs.mutex.Lock()
+	defer cs.mutex.Unlock()
+	for k, v := range cs.connections {
+		err := callback(k, v)
+		if err != nil {
+			klog.Errorf("Callback execution failed for flow with key: %v, conn: %v, k, v: %v", k, v, err)
+			return err
 		}
 	}
+	return nil
+}
+
+// DeleteConnWithoutLock deletes the connection from the connection map given
+// the connection key without grabbing the lock. Caller is expected to grab lock.
+func (cs *connectionStore) DeleteConnWithoutLock(connKey flowexporter.ConnectionKey) error {
+	_, exists := cs.connections[connKey]
+	if !exists {
+		return fmt.Errorf("connection with key %v doesn't exist in map", connKey)
+	}
+	delete(cs.connections, connKey)
+	if !cs.isDenyConn {
```

Comment:
could we use the tag from the connection instead of connection store? Or have a separate individual methods for decrementing metrics and call it along with `DeleteConnWithoutLock`.